lib/combine_pdf/pdf_public.rb

```diff
@@ -95,7 +95,14 @@ def initialize(parser = nil)
       @info = {}
       parser ||= PDFParser.new('')
       raise TypeError, "initialization error, expecting CombinePDF::PDFParser or nil, but got #{parser.class.name}" unless parser.is_a? PDFParser
-      @objects = parser.parse
```

Comment:
Is this catch-and-release logic actually required?

If the file is encrypted we will get the correct exception, whereas if the `Zlib::DataError` is raised for a different reason we will now get the wrong exception (making it more difficult to debug).

Reply:
@boazsegev Agreed ☝️ 

Maybe you can help the author to revert this part and then tweak the minitest to get in merge sooner? We have a pressing use case that requires ingesting this flag and I would rather not to create a fork if possible.

Reply:
You guys are free to do whatever you'd like with this PR. I don't even think we use this any longer for this purpose, so change whatever is needed.

Reply:
@leviwilson I don't have permission. 

It will be very cool if you can give  permission to your branch temporary to push up a commit?

Reply:
@kimyu92 send me the patch and I can apply it

Reply:
Here is the patch. 

 ```diff
 diff --git a/lib/combine_pdf/pdf_public.rb b/lib/combine_pdf/pdf_public.rb
index 41261e3..8784d21 100644
--- a/lib/combine_pdf/pdf_public.rb
+++ b/lib/combine_pdf/pdf_public.rb
@@ -95,13 +95,7 @@ module CombinePDF
       @info = {}
       parser ||= PDFParser.new('')
       raise TypeError, "initialization error, expecting CombinePDF::PDFParser or nil, but got #{parser.class.name}" unless parser.is_a? PDFParser
-
-      begin
-        @objects = parser.parse
-      rescue Zlib::DataError => e
-        raise CombinePDF::EncryptionError, 'the file is encrypted' if parser.root_object[:Encrypt] && parser.raise_on_encrypted
-        raise e
-      end
+      @objects = parser.parse
 
       # remove any existing id's
       remove_old_ids
diff --git a/test/combine_pdf/load_test.rb b/test/combine_pdf/load_test.rb
index 4305871..47ee57a 100644
--- a/test/combine_pdf/load_test.rb
+++ b/test/combine_pdf/load_test.rb
@@ -9,51 +9,39 @@ describe 'CombinePDF.load' do
   subject { CombinePDF.load "test/fixtures/files/#{file}", options }
 
   describe 'raise_on_encrypted option' do
-    let(:raise_on_encrypted) { false }
     let(:file) { 'sample_encrypted_pdf.pdf' }
     let(:options) { { raise_on_encrypted: raise_on_encrypted } }
 
-    it('has a PDF') { assert_instance_of CombinePDF::PDF, subject }
-
-    describe 'raise_on_encrypted: true' do
+    describe 'when raise_on_encrypted: true' do
       let(:raise_on_encrypted) { true }
 
-      it('raises an CombinePDF::EncryptionError') do
-        error = assert_raises(CombinePDF::EncryptionError) { subject }
-        assert_match 'the file is encrypted', error.message
+      describe 'with encrypted file' do
+        it('raises an CombinePDF::EncryptionError') do
+          error = assert_raises(CombinePDF::EncryptionError) { subject }
+          assert_match 'the file is encrypted', error.message
+        end
       end
 
-      describe 'non-encrypted files' do
+      describe 'with unencrypted file' do
         let(:file) { 'sample_pdf.pdf' }
 
         it('has a PDF') { assert_instance_of CombinePDF::PDF, subject }
       end
+    end
 
-      describe 'Zlib::DataError' do
-        let(:encrypted?) { false }
-        let(:parser) { Minitest::Mock.new }
-
-        around do |example|
-          parser.expect(:parse, true) { raise Zlib::DataError, 'incorrect data header' }
-          root_object = encrypted? ? { Encrypt: :yes } : { NotEncrypted: :cool }
-          parser.expect(:root_object, root_object)
-          parser.expect(:raise_on_encrypted, raise_on_encrypted)
-          parser.expect(:is_a?, true, [CombinePDF::PDFParser])
-          CombinePDF::PDFParser.stub(:new, parser) { |_| example.call }
-        end
+    describe 'when raise_on_encrypted: false' do
+      let(:raise_on_encrypted) { false }
 
-        it('raises Zlib::DataError') do
-          assert_raises(Zlib::DataError) { subject }
+      describe 'with encrypted file' do
+        it('does not raise an CombinePDF::EncryptionError') do
+          assert_instance_of CombinePDF::PDF, subject
         end
+      end
 
-        describe 'encrypted' do
-          let(:encrypted?) { true }
+      describe 'with unencrypted file' do
+        let(:file) { 'sample_pdf.pdf' }
 
-          it('raises an CombinePDF::EncryptionError') do
-            error = assert_raises(CombinePDF::EncryptionError) { subject }
-            assert_match 'the file is encrypted', error.message
-          end
-        end
+        it('has a PDF') { assert_instance_of CombinePDF::PDF, subject }
       end
     end
   end
```

Reply:
@kimyu92 not sure why, but `git apply <your patch>` is not working for me

```
$  git apply ~/combine_pdf.patch
error: patch failed: lib/combine_pdf/pdf_public.rb:95
error: lib/combine_pdf/pdf_public.rb: patch does not apply
error: patch failed: test/combine_pdf/load_test.rb:9
error: test/combine_pdf/load_test.rb: patch does not apply
```

Reply:
@leviwilson i have tested directly against your branch and it should be working though.

```sh
git clone git@github.com:leviwilson/combine_pdf.git
git checkout raise_on_encrypted
git apply kim.patch
```

[kim.patch](https://github.com/boazsegev/combine_pdf/files/11141259/kim.patch)

Reply:
The file version you attached worked for me, thanks. The copy / paste into a `combine_pdf.patch` didn't work.

I updated the branch.

Reply:
@boazsegev Let us know how else we can push through this PR. 🙂